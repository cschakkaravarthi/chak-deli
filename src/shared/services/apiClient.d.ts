declare const client: import("axios").AxiosInstance;
export default client;
//# sourceMappingURL=apiClient.d.ts.map