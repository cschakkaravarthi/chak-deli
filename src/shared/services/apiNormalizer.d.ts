import Article from '../models/Article.model';
import Bookmark from '../models/Bookmark.model';
import BoxModel from '../models/Box.model';
import EventModel from '../models/Event.model';
import Link, { LinkGroupItems } from './../models/Link.model';
import Notification from '../models/Notification.model';
import OfficesPeople, { OfficeItem } from './../models/OfficesPeople.model';
import PeopleModel from '../models/People.model';
import Todo from '../models/Todo.model';
import { CatalogSearch } from '../models/Catalog.model';
import { ContactGroup, Department } from '../models/Department.model';
import { DocumentSearch } from '../models/Document.model';
import { KnowledgeSearch } from '../models/KnowledgeBase.model';
import { Manager, Users } from '../models/OrgChart.model';
import { Office, OfficeSearch } from '../models/Office.model';
import { Page, PageSearch } from '../models/Page.model';
import { ResponseData, SuccessfulResponse } from './apiService';
import { Categories, ManualNotification, ManualNotificationListModel, ManualNotifications, NotificationBell } from '../models/ManualNotification.model';
import UserInfoModel, { UserProfileModel, UserProfilePicture, ContactList, UserProfileUpdateResponse, ContactResponse } from '../../shared/models/UserInfo.model';
import { TaxonomyTermModel, VariousContentGroupModel, VariousContentModel } from '../types/contentTypes';
export default class ApiNormalizer {
    /**
     * Sanitation and validation of API response is crucial to prevent bugs and XSS.
     */
    static normalizeContent(response: ResponseData): VariousContentModel[];
    static normalizeGroupedContent(response: ResponseData): VariousContentGroupModel[];
    static normalizeContentSetGroup(response: ResponseData): VariousContentGroupModel | null;
    static normalizeContentSet(response: ResponseData): VariousContentModel | null;
    static validateArticle(content: ResponseData): ResponseData | null;
    static validateEvent(content: ResponseData): ResponseData | null;
    static validateDepartment(content: ResponseData): ResponseData | null;
    static validateOffice(content: ResponseData): ResponseData | null;
    static validatePeople(content: ResponseData): ResponseData | null;
    static validateNotification(content: ResponseData): ResponseData | null;
    static validateLinks(content: ResponseData): ResponseData | null;
    static validateBox(content: ResponseData): ResponseData | null;
    static validateSuccessfulResponse(content: ResponseData): ResponseData | null;
    static normalizeMinisiteLink(untrustedContent: ResponseData): Link | null;
    static normalizeSuccessfulResponse(untrustedContent: ResponseData): SuccessfulResponse | null;
    static normalizeBox(untrustedContent: ResponseData): BoxModel | null;
    /**
     * Convert API response into the data model we want to store in state.
     */
    static normalizeArticle(untrustedContent: ResponseData): Article | null;
    static normalizeTaxonomyTerm(content: ResponseData): TaxonomyTermModel | null;
    static normalizeTaxonomyTerms(content: ResponseData): TaxonomyTermModel[];
    static normalizeEvent(untrustedContent: ResponseData): EventModel | null;
    static normalizeGroupedContacts(contactsString: string): ContactGroup[];
    static normalizeDepartment(untrustedContent: ResponseData): Department | null;
    static normalizePerson(untrustedContent: ResponseData): PeopleModel | null;
    static validatePageSearch(content: ResponseData): ResponseData | null;
    static normalizePageSearch(untrustedContent: ResponseData): PageSearch | null;
    static validateOfficeSearch(content: ResponseData): ResponseData | null;
    static normalizeOfficeSearch(untrustedContent: ResponseData): OfficeSearch | null;
    static validateDocumentSearch(content: ResponseData): ResponseData | null;
    static normalizeDocumentSearch(untrustedContent: ResponseData): DocumentSearch | null;
    static validateKnowledgebaseSearch(content: ResponseData): ResponseData | null;
    static normalizeKnowledgebaseSearch(untrustedContent: ResponseData): KnowledgeSearch | null;
    static validateCatalogSearch(content: ResponseData): ResponseData | null;
    static normalizeCatalogSearch(untrustedContent: ResponseData): CatalogSearch | null;
    static normalizeOffice(untrustedContent: ResponseData): Office | null;
    static normalizeSearch(response: ResponseData): ResponseData;
    static normalizeSearchSuggestions(response: ResponseData): ResponseData;
    static normalizeNotification(untrustedContent: ResponseData): Notification | null;
    static normalizeManualNotification(untrustedContent: ResponseData): ManualNotification | null;
    static validateManualNotification(content: ResponseData): ResponseData | null;
    static dataToObjectManualNotification(contentData: ResponseData): ManualNotification;
    static normalizeManualNotificationFilterList(content: ResponseData): Categories[];
    static normalizeManualNotificationFilters(content: ResponseData): Categories[];
    static normalizeManualNotificationFilterData(untrustedContent: ResponseData): Categories | null;
    static validateManualNotificationFilterList(content: ResponseData): ResponseData | null;
    static normalizeArticles(response: ResponseData): ResponseData;
    static normalizeEvents(response: ResponseData): ResponseData;
    static normalizeFacets(responseFacets: ResponseData): ResponseData;
    static normalizeDepartments(response: ResponseData): Department[];
    static normalizeLink(untrustedContent: ResponseData): LinkGroupItems | null;
    static normalizeQuickLink(untrustedContent: ResponseData): LinkGroupItems | null;
    static normalizeTodo(untrustedContent: ResponseData): Todo | null;
    static normalizeBookmark(untrustedContent: ResponseData): Bookmark | null;
    static normalizeMenuLinks(content: ResponseData): LinkGroupItems[];
    static normalizeUserLinks(trustedContent: ResponseData): string[];
    static normalizeTodos(trustedContent: ResponseData): Todo[];
    static normalizeBookmarks(trustedContent: ResponseData): Bookmark[];
    static normalizePeople(content: ResponseData): PeopleModel[];
    static normalizeManualNotificationList(content: ResponseData): ManualNotificationListModel;
    static normalizeManualNotifications(content: ResponseData): ManualNotifications;
    static normalizeNotificationsBell(content: ResponseData): NotificationBell | null;
    static normalizeNotificationBell(untrustedContent: ResponseData): NotificationBell | null;
    static validateNotificationBell(content: ResponseData): ResponseData | null;
    static normalizeNotificationsArchive(content: ResponseData): boolean | null;
    static normalizeNotificationArchive(untrustedContent: ResponseData): boolean | null;
    static validateNotificationArchive(content: ResponseData): ResponseData | null;
    static normalizeArticleById(content: ResponseData): Article[];
    static normalizeEventById(content: ResponseData): EventModel[];
    static validateUser(content: ResponseData): ResponseData | null;
    static userAndProfileContent(untrustedContent: ResponseData): UserInfoModel | null;
    static userProfilePictureContent(untrustedContent: ResponseData): UserProfilePicture | null;
    static userProfileContent(untrustedContent: ResponseData): UserProfileModel | null;
    static normalizeUser(untrustedContent: ResponseData): UserInfoModel | null;
    static normalizeProfile(untrustedContent: ResponseData): UserInfoModel | null;
    static normalizeUsers(content: ResponseData): UserInfoModel[];
    static normalizePage(response: ResponseData): Page | null;
    static normalizeUserProfile(untrustedContent: ResponseData): UserProfileModel | null;
    static normalizeUserProfilePicture(untrustedContent: ResponseData): UserProfilePicture | null;
    static normalizeOrgChart(content: ResponseData): Users;
    static normalizeSubordinates(untrustedContent: ResponseData): Manager | null;
    static validateSubordinates(content: ResponseData): ResponseData | null;
    static normalizeOfficesPeoples(content: ResponseData): OfficesPeople[];
    static normalizeOfficesPeople(untrustedContent: ResponseData): OfficesPeople | null;
    static normalizeOfficeContents(content: ResponseData): OfficeItem[];
    static normalizeSetUserProfileData(content: any): any;
    static normalizeOfficeContent(content: ResponseData): OfficeItem | null;
    static validateOfficesPeople(content: ResponseData): ResponseData | null;
    static validateOfficeContent(content: ResponseData): ResponseData | null;
    static normalizeUserProfileUpdate(content: ResponseData): UserProfileUpdateResponse;
    static normalizeMyContacts(content: ResponseData): ContactList | null;
    static normalizeMyContactsResponse(content: ResponseData): ContactList[];
    static normalizeUsersByEmail(content: ResponseData): UserInfoModel[];
    static normalizeProfilepicDeleteResponse(content: ResponseData): ContactResponse | null;
}
//# sourceMappingURL=apiNormalizer.d.ts.map