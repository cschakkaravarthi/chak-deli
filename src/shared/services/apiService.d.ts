import Article from '../models/Article.model';
import EventModel from '../models/Event.model';
import Link, { LinkGroupItems } from '../models/Link.model';
import Notification from '../models/Notification.model';
import PeopleModel from '../models/People.model';
import UserInfoModel, { UserProfileModel, UserProfilePicture, ContactResponse, ContactList, UserProfileUpdateResponse, AddToContactResponse } from '../models/UserInfo.model';
import { ManualNotificationListModel, NotificationBell } from '../models/ManualNotification.model';
import { ContentWithFacet } from '../types/contentWithFacets';
import { Department } from '../models/Department.model';
import { Office } from '../models/Office.model';
import OfficesPeople from '../models/OfficesPeople.model';
import { Page } from '../models/Page.model';
import { SearchResultState } from '../types/searchResultTypes';
import { Users } from '../models/OrgChart.model';
import Bookmark from '../models/Bookmark.model';
import Todo from '../models/Todo.model';
import { VariousContentGroupModel, VariousContentModel } from '../types/contentTypes';
export interface ResponseData {
    [key: string]: any;
}
export interface SuccessfulResponse {
    success: boolean;
}
export default class ApiService {
    static makeApiCall: <T>(url: string, normalizeCallback: <T_1>(d: ResponseData) => any, callback: (d: T) => any, errorCallBack?: ((a: string) => any) | undefined, method?: string, data?: {}, header?: {
        'Content-Type': string;
    }) => Promise<void>;
    static getContent: (callback: (groupedContent: VariousContentModel[]) => void, filterId: string, filterField?: string | undefined, allowedTypes?: string | undefined, sort?: string | undefined) => Promise<void>;
    static getGroupedContent: (callback: (groupedContent: VariousContentGroupModel[]) => void, filterId: string, filterField?: string | undefined, allowedTypes?: string | undefined, group?: string | undefined, sort?: string | undefined) => Promise<void>;
    static getArticles: (callback: (a: ContentWithFacet) => void, page?: number | undefined, limit?: number | undefined, sort?: string | undefined, facet?: string | undefined, selectedFacet?: string | undefined) => Promise<void>;
    static getPeople: (callback: (a: PeopleModel[]) => void, sitecode: string, startsWith?: string | undefined, page?: string | number | undefined, limit?: string | number | undefined) => Promise<void>;
    static getEvents: (callback: (a: ContentWithFacet) => void, page?: number | undefined, limit?: number | undefined, sort?: string | undefined, facet?: string | undefined, selectedFacet?: string | undefined) => Promise<void>;
    static makeApiCallWithoutCallback: (url: string, errorCallBack?: ((a: string) => any) | undefined) => Promise<any>;
    static getArticleById: (callback: (a: Article[]) => void, id: string) => Promise<void>;
    static getEventById: (callback: (a: EventModel[]) => void, id: string) => Promise<void>;
    static getNotifications: (callback: (a: Notification[]) => void) => Promise<void>;
    static getManualNotifications: (callback: (a: ManualNotificationListModel) => void, id: string) => Promise<void>;
    static getManualNotificationsFilterList: (callback: (a: import("../types/contentTypes").TaxonomyTermModel[]) => void) => Promise<void>;
    static getNotificationsBell: (callback: (a: NotificationBell) => void) => Promise<void>;
    static setNotificationsArchive: (callback: (success: boolean) => void, errorback: (error: boolean) => void, drupal_id: number | undefined) => Promise<void>;
    static getUserByNetworkId: (callback: (a: UserInfoModel[]) => void, userID: string) => Promise<void>;
    static getUserByEmail: (callback: (a: UserInfoModel[]) => void, email: string) => Promise<void>;
    static getProfile: (callback: (a: UserInfoModel) => void) => Promise<void>;
    static getMenuLinks: (callback: (a: LinkGroupItems[]) => void) => Promise<void>;
    static getUserLinks: (callback: (userLinksIds: Link[]) => void) => Promise<void>;
    static getTodos: (callback: (todos: Todo[]) => void) => Promise<void>;
    static setTodo: (callback: (success: boolean) => void, text: string) => Promise<void>;
    static deleteTodo: (callback: (success: boolean) => void, identifier: string) => Promise<void>;
    static getBookmarks: (callback: (bookmarks: Bookmark[]) => void) => Promise<void>;
    static setBookmark: (callback: (success: boolean) => void, title: string, url: string) => Promise<void>;
    static deleteBookmark: (callback: (success: boolean) => void, identifier: string) => Promise<void>;
    static getMinisiteLinks: (callback: (minisiteLinks: VariousContentGroupModel[]) => void, cat: string) => Promise<void>;
    static getPages: (callback: (pages: Page[]) => void, categoryId?: string | null) => Promise<void>;
    static getPage: (callback: (pages: Page) => void, pageId: number) => Promise<void>;
    static getDepartment: (callback: (department: Department) => void, departmentId: number) => Promise<void>;
    static setUserLinks: (callback: (success: boolean) => void, links: string[]) => Promise<void>;
    static getHomeArticles: (callback: (a: ResponseData) => void, limit: string, sort: string, owner?: string | undefined) => Promise<void>;
    static getHomeEvents: (callback: (a: ResponseData) => void, owner?: string | undefined) => Promise<void>;
    static getSearchResults: (callback: (a: SearchResultState) => void, errorCallback: (a: string) => void, page?: number | undefined, query?: string | undefined, limit?: number | undefined, type?: string | undefined, facet?: string | undefined) => Promise<void>;
    static getSearchSuggestions: (callback: (a: ResponseData) => void, query?: string | undefined, limit?: number | undefined) => Promise<void>;
    static getUserProfile: (callback: (a: UserProfileModel) => void, errorCallback: (a: string) => void, username?: string | undefined) => Promise<void>;
    static getUserProfilePicture: (callback: (a: UserProfilePicture) => void, errorCallback: (a: string) => void, username?: string | undefined) => Promise<void>;
    static getOffice: (callback: (a: Office) => void, errorCallback: (a: string) => void, officeId?: string | undefined) => Promise<void>;
    static getFacilities: (callback: (a: Link[]) => void, errorCallback: (a: string) => void, categoryId?: string | undefined) => Promise<void>;
    static getUserOrgChart: (callback: (a: Users) => void, errorCallback: (a: string) => void, email: string) => Promise<void>;
    static downloadAddToCalendarService: (id: string) => Promise<any>;
    static getEmployeeServices: (callback: (a: VariousContentModel[]) => void) => Promise<void>;
    static getDepartments: (callback: (a: Department[]) => void) => Promise<void>;
    static getQuickTools: (callback: (a: Link[]) => void) => Promise<void>;
    static getOfficesPeople: (callback: (a: OfficesPeople[]) => void) => Promise<void>;
    static toggleContentLike: (callback: (a: ResponseData) => void, contentType: string, contentId: number, likedByUser: boolean) => Promise<void>;
    static updateUserProfileDetails: (callback: (a: UserProfileUpdateResponse) => void, errorCallback: (a: string) => void, userDetails: Pick<UserProfileModel, "email" | "photo" | "error" | "networkID" | "department" | "profilePicture" | "lastName" | "firstName" | "fullName" | "jobLocation" | "companyName" | "profitCenter" | "desk" | "officeNumber" | "mobileNumber" | "nickname" | "birthday" | "birthMonth" | "anniversaryDay" | "anniversaryMonth" | "languagesSpoken" | "professionalSkills" | "currentlyWorking" | "aboutMe" | "hideMobileNumberOnProfile" | "avatar">) => Promise<void>;
    static postAddToContact: (callback: (response: AddToContactResponse) => void, errorCallback: () => void, email: string | undefined) => Promise<void>;
    static fetchMyContactsList: (callback: (a: ContactList) => void, errorCallback: (a: string) => void) => Promise<void>;
    static removeContactFromList: (callback: (a: ContactResponse) => void, errorCallback: (a: string) => void, email?: string | undefined) => Promise<void>;
    static uploadUserProfilePic: (callback: (a: ContactResponse) => void, errorCallback: (a: string) => void, file: File) => Promise<void>;
    static removeUserProfilePic: (callback: (a: ContactResponse) => void, errorCallback: (a: string) => void) => Promise<void>;
}
//# sourceMappingURL=apiService.d.ts.map