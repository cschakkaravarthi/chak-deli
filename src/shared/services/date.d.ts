export declare const formatDate: (unixTimestamp?: number | undefined, dateTime?: string | undefined, format?: string) => string;
/**
 * Return Event Start Time if duration is 23:59:00 or less
 * hours * mints = 24 * 60 = 1440 - 1 mint = 1430
 * @param unixTimestampDuration
 * @param unixTimestampStartTime
 * @param format
 */
export declare const getEventTime: (unixTimestampDuration: number, unixTimestampStartTime: number, format?: string) => string;
export declare const formatDateTime: (unixTimestamp?: number | undefined, format?: string) => string;
export declare const formatDateRange: (startDate: number, endDate: number) => string;
export declare const formatDateRangeWithTime: (startDate: number, endDate: number) => string;
export declare const currentDate: (format?: string) => string;
interface ValidateDateErrorMessage {
    validDateError: string;
    bothFieldError: string;
}
export declare const validateDate: (day: string, month: string, message: ValidateDateErrorMessage) => string;
export {};
//# sourceMappingURL=date.d.ts.map