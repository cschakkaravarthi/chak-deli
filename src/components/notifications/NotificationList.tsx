import Notification from '../../shared/models/Notification.model';
import React, { FC } from 'react';
import NotificationTeaser from './NotificationTeaser';
import ListGroup from 'react-bootstrap/ListGroup';

interface Props {
  notifications?: Notification[];
}

const NotificationList: FC<Props> = props => {
  if (!props.notifications) return null;

  return (
    <ListGroup variant='flush'>
      {props.notifications.map(n => (
        <ListGroup.Item key={n.id}>
          <NotificationTeaser notification={n} />
        </ListGroup.Item>
      ))}
    </ListGroup>
  );
};

export default NotificationList;
