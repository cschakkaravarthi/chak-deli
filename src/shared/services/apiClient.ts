import axios from 'axios';
import get from 'lodash.get';
import packageJson from '../../../package.json';
import { getAccessToken } from '../../reactAuthProvider';
import { appEnv } from '../../utils/customHooks';

const apiVersion = packageJson.umgcApi.useEdge ? 'dev/' : `v${packageJson.umgcApi.version}/`;

// For some reason appEnv is undefined at this point, but window.env is not?
const client = axios.create({
  baseURL: get(appEnv || window.env, 'centralApi.baseUrl') + apiVersion
});

client.interceptors.request.use(request => {
  return new Promise((resolve, reject) => {
    getAccessToken()
      .then(accessTokenRepsonse => {
        request.headers.Authorization = 'Bearer ' + accessTokenRepsonse;
        resolve(request);
      })
      .catch(() => {
        // console.log('the catch token is ' + client.defaults.headers.common.Authorization);
        reject(request);
      });
  });
});

// @TODO: Logs for dev only.
client.interceptors.request.use(request => {
  // console.log('Starting Request:');
  // console.log(request);
  return request;
});

client.interceptors.response.use(response => {
  // console.log('Response:');
  // console.log(response);
  return response;
});

export default client;
