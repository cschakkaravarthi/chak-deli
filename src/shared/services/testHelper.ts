import { ReactElement } from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import getStore from './mockGlobalStore';

export const shallowRender = (toRender: JSX.Element): ReactElement => {
  const renderer = ShallowRenderer.createRenderer();
  renderer.render(toRender);
  return renderer.getRenderOutput();
};

export const getMockStore = getStore;

export const drawerMockNavigation = {
  state: {
    isDrawerOpen: true,
    isTransitioning: false,
    index: 1,
    routes: [],
    key: 'abc',
    params: {}
  },
  dispatch: jest.fn(),
  goBack: jest.fn(),
  dismiss: jest.fn(),
  navigate: jest.fn(),
  openDrawer: jest.fn(),
  closeDrawer: jest.fn(),
  toggleDrawer: jest.fn(),
  getParam: jest.fn(),
  setParams: jest.fn(),
  addListener: jest.fn(),
  push: jest.fn(),
  replace: jest.fn(),
  reset: jest.fn(),
  pop: jest.fn(),
  popToTop: jest.fn(),
  isFocused: jest.fn(),
  isFirstRouteInParent: jest.fn(),
  dangerouslyGetParent: jest.fn()
};
