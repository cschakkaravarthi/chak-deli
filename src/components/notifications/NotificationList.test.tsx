import React from 'react';
import { dummyNotifications } from '../../shared/models/Notification.model';
import { shallowRender } from '../../shared/services/testHelper';
import NotificationList from './NotificationList';

describe('NotificationList', () => {
  it('renders null without notifications', () => {
    const tree = shallowRender(<NotificationList />);
    expect(tree).toEqual(null);
  });

  it('renders correctly with notifications', () => {
    const tree = shallowRender(
      <NotificationList notifications={dummyNotifications} />
    );
    expect(tree).toMatchSnapshot();
  });
});
