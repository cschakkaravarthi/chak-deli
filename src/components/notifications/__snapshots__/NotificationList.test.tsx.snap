// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotificationList renders correctly with notifications 1`] = `
<ForwardRef
  horizontal={null}
  variant="flush"
>
  <ForwardRef
    active={false}
    disabled={false}
    variant={null}
  >
    <NotificationTeaser
      notification={
        Object {
          "application": "Deep Thought",
          "approve": true,
          "description": "Answer to the Ultimate Question of Life, the Universe, and Everything",
          "detailsLink": "http://google.com/",
          "id": 42,
          "reject": true,
          "viewed": true,
        }
      }
    />
  </ForwardRef>
  <ForwardRef
    active={false}
    disabled={false}
    variant={null}
  >
    <NotificationTeaser
      notification={
        Object {
          "application": "Deep Thought",
          "approve": true,
          "description": "Answer to the Ultimate Question of Life, the Universe, and Everything",
          "detailsLink": "http://google.com/",
          "id": 24,
          "reject": true,
          "viewed": true,
        }
      }
    />
  </ForwardRef>
</ForwardRef>
`;
